packages/graph/hash_graph/bin/hash_graph/src/main.rs

```diff
@@ -254,3 +237,52 @@ async fn main() -> Result<(), GraphError> {
 
     Ok(())
 }
+
+async fn migrate_database(_args: Args, pool: PostgresStorePool<NoTls>) -> Result<(), GraphError> {
+    let mut connection = pool
+        .acquire()
+        .await
+        .change_context(GraphError)
+        .map_err(|err| {
+            tracing::error!("{err:?}");
+            err
+        })?;
+
+    connection
+        .run_migrations()
+        .await
+        .change_context(GraphError)
+        .map_err(|err| {
+            tracing::error!("{err:?}");
+            err
+        })?;
+
+    Ok(())
+}
+
+#[tokio::main]
+async fn main() -> Result<(), GraphError> {
+    let args = Args::parse_args();
+
+    let log_args = args.log_config.clone();
+    let _log_guard = init_logger(
+        log_args.log_format,
+        log_args.log_folder,
+        log_args.log_level,
+        &log_args.log_file_prefix,
+        args.otlp_endpoint.as_deref(),
+    );
+
+    let pool = PostgresStorePool::new(&args.db_info, NoTls)
+        .await
+        .change_context(GraphError)
+        .map_err(|err| {
+            tracing::error!("{err:?}");
+            err
+        })?;
+
+    match args.subcommand() {
+        args::Subcommand::Server => start_server(args, pool).await,
+        args::Subcommand::Migrate => migrate_database(args, pool).await,
+    }
```

Comment:
Can we have a file per subcommand?

Reply:
Yep, I'll look into that with the split of of the subcommand args

Reply:
cb75b3c7931ddd714f052104085287fa7e650fcf & 573215c31cd3d050370e525d330d9612890a156a & 84b6eb6b6d3798983cb85d4f3353e687e80b87a2
thank you!